### ISSUE TITLE
Moving Solar Hot Water Components Out of WIP

### ISSUE THREAD

**Issue by chriswmackey** [mostaphaRoudsari/ladybug#278]
@stgeorges , @mostaphaRoudsari and any other users of the PV/Hot water components,

I am opening an issue here to discuss whether and (if so) how the solar hot water components should be moved out of WIP for the coming release in a few days.  This said, I don't think that we necessarily have to let the pressure of the release drive this issue and we can also just as easily do the release and then make these changes soon after.  If we are close to having everything ready, we might as well just make it a part of the release, though.

In honesty, I have not gotten the chance to test the hot water components much but I have the sense that they have been there long enough with enough pull requests / bug fixes to the core renewables components over the last few months that they can emerge from the WIP section. @stgeorges , please confirm whether you think my sense is correct.  I also noticed that you had an improved version of the TOF component in one of the example files you sent me on another issue and, if you have a timeline for another version of this component, please let us know here.

As far as what will happen once we move the hot water components out, I could imagine 2 scenarios:

**OPTION 1** -  We load them all into the same Panel 4 of the EnvironmentalAnalysis Tab.  The less critical components to only show up in the drop-down and/or in a 5th panel on the EnvironmentalAnalysis tab.

**OPTION 2** - 15 components really seems sufficient to have a completely separate tab in Ladybug and so we could add a 4 | Renewables tab.  This would cause Ladybug to  follow the structure of the [plugin diagrams](https://docs.google.com/presentation/d/1e_XmZj31HTAKlYL0SCmzD8q2hmjLXSnKVHeBBrw8rpw/edit#slide=id.g11504c007a_0_3) with 4 core capabilities in total.  In this case, I think we would fully defer to your judgement @stgeorges about how to structure the panels and location of the components within this new tab.

Of these two options, I am leaning more towards option 2 with a new Renewables tab but let me know if you feel otherwise or if there are other options that I have not listed here.

Looking forward to the discussion!

**Comment by stgeorges** [comment]
Hi @chriswmackey , @mostaphaRoudsari ,
Thank you for opening the discussion.

```Option 1``` - I think it would be better if Photovoltaics components will be in EnvironmentalAnalysis Tab, Panel 4 while all Solar water heating ones would be in Panel 5.

```Option 2``` - This is also a good idea.

The thing that I do not like about ```Option 2``` is that it may require separating all Photovoltaics and Solar water heating components into a couple of panels. While this may look like a good thing (first panel might be reserved for main components, second panel for components corresponding to losses, third panel for shading components...) I am worrying about the mixing of the Photovoltaics and Solar water heating.
Even though both use "Sunpath shading" component to account for shading, in my mind they somehow need to be located in separate panels.
Maybe this can be solved by having only two panels in the "4 | Renewables tab". First one for Photovoltaics, second one for Solar water heating?

I will welcome suggestions.

**Comment by chriswmackey** [comment]
@stgeorges ,

Good to know about the TOF component.  Let's proceed with the plan to do it before the release.

There is certainly no pressure to have more than two panels in a "4 | Renewables tab" scenario.  You know your components best and I defer to your judgement.  The only thing that I think the Renweables tab offers that the other option is not so good with is that you can put components that are not critical to most workflows only in the dropdown of the tab (for example the Photovoltaics Module Settings might go in the dropdown).  We could, in theory, do the same with the 3 | Environmental Analysis tab but your components will be next to the raytracing and simple solar envelope / fan, which could make it less obvious to users how the components are intended to work together.

I agree horizon angles belongs with the PV/Hot Water components.  I also agree that terrain shading mask is a broadly-applicable component and should not be in the renewables tab.  We both saw in the forums that some people have started using it for energy simulations and it has clear applications in radiation studies, etc.  All of this said, I would be more comfortable keeping the terrain shading mask component in WIP for just a bit while longer.  This is because of the newness (1 month), complexity, and broad applicability of it.  I know that I have to get around to it at some point but I think there are some options I and/or we can add to make it more optimized for energy simulations to say the least.  Let me know if you agree.

-Chris

**Comment by stgeorges** [comment]
Hi @chriswmackey ,

>Good to know about the TOF component. Let's proceed with the plan to do it before the release.

I didn't understand the upper last sentence. What should be done before the release?

>"The only thing that I think the Renweables tab offers that the other option is not so good with is that you can put components that are not critical to most workflows only in the dropdown of the tab

I didn't understand you this part.

>All of this said, I would be more comfortable keeping the terrain shading mask component in WIP for just a bit while longer. This is because of the newness (1 month), complexity, and broad applicability of it. I know that I have to get around to it at some point but I think there are some options I and/or we can add to make it more optimized for energy simulations to say the least. Let me know if you agree.

Ok.

**Comment by chriswmackey** [comment]
@stgeorges ,

All that we have to do before the release is decide where to put the components and put them there.  Unless there are any known bugs in the components which we should solve before the release (it sounds like there aren't any currently).

To clarify, there are a number of Ladybug components that only appear in the toolbar when you click to drop down a tab.  These components never show up in the tabs themselves (only when you click on the tab in the menu that drops down). 
![image](https://cloud.githubusercontent.com/assets/5567574/17459291/be671572-5bff-11e6-800d-d77567d6633e.png)
 The 4 | Extra tab has a lot of these and an example is the Ladybug_Oreintation Study Parameters.  You will notice that these components contain the following code:
`ghenv.Component.AdditionalHelpFromDocStrings = "0"`
 Putting components that are not as important in this "0" category is helpful so that beginner users focus first on the components that they can see in the tab and, when they become advanced users, they might use the components int he "0" category to customize their workflows.

Let me know if this is clear.

**Comment by stgeorges** [comment]
Thank you for the screenshot and explanation Chris! It is more clear now.

I am using a 15" monitor with 1024x768 resolution, so the number of components visible to my screen are definitely smaller. I see now from your screenshot how stretched the whole components toolbar can be.

But putting some of the Photovoltaics and Solar water heating components into the "0" category would mean that they will be mixed in that category which is what I have been trying to avoid in the first place.

I somehow prefer putting all Photovoltaics components into the "1" category, and all Solar water heating into the category "2" of the "4 | Renewables" tab.

**Comment by chriswmackey** [comment]
@stgeorges ,

Understood.  I trust your judgement about the best arrangement for your components and, if 2 panels with nothing below is best, then let's go for it.

I have finished editing everything in the Ladybug code base for the upcoming release.  At this point, you can send a pull request with your ideal arrangement of the components, I can merge it in without conflicts.

I sense that we would do the release in another couple of days so please send the pull request before then if possible.

-Chris

**Comment by chriswmackey** [comment]
@stgeorges ,

I just wanted to post this old conversation as it seems relevant:
http://www.grasshopper3d.com/group/ladybug/forum/topics/ladybug-sunrise-sunset?commentId=2985220%3AComment%3A1463246&groupId=2985220%3AGroup%3A658987

At present, one of the only ways we have to address Tim's and similar user's concern is by organizing components in a hierarchy with the most important as the most visible.  Again, the call is yours and I support the 2-panel organization if you think it's best but I thought that it is good to at least consider the opposing argument before deciding.

-Chris

**Comment by stgeorges** [comment]
Hi @chriswmackey ,

Now that ```4 | Renewables``` sub-category (tab) is suppose to be introduced, does that mean that ```"6 | WIP"``` category, will become ```"7 | WIP"```?

>I just wanted to post this old conversation as it seems relevant:
http://www.grasshopper3d.com/group/ladybug/forum/topics/ladybug-sunrise-sunset?commentId=2985220%3AComment%3A1463246&groupId=2985220%3AGroup%3A658987

>At present, one of the only ways we have to address Tim's and similar user's concern is by organizing components in a hierarchy with the most important as the most visible. Again, the call is yours and I support the 2-panel organization if you think it's best but I thought that it is good to at least consider the opposing argument before deciding.

Thank you for posting the link to that discussion.
As far as I understood, Tim has been complaining about the number of Honeybee components. At the moment there are some 50% more of them than Ladybug component which is a significant number. I highly doubt that this ratio was drastically less in February this year, when those comments were made.
But I also understand that this may be applicable to Ladybug components as well.
There are 7 Photovoltaics and 7 Solar water heating components. I think dividing them into more than 2 panels would exactly contribute to more confusion.
In my opinion, the best approach to comply with your multiple panels would be to have the panel 1 being reserved for only two main components: "Photovoltaics Surface" and "Solar water heating Surface", then put all of the other Photovoltaics components into the panel 2, and all of the other Solar water heating components into the panel 3.
If this is ok with you, I can send a pull request.

**Comment by chriswmackey** [comment]
@stgeorges ,

In relation to Tim's comments, your logic sounds good and I buy your argument.

The change in categories that you mention is correct but, in your pull request, you only have to move your components into a 4 | Renewables tab and I can take care of moving the rest over.

I think that we may do the release relatively soon so, the sooner that you send the request, the better.  I can also move them as I have a pretty strong sense of how to organize them in 2 categories based on what you have said here.  Still, it would be better if you did it just for the github record.

-Chris

**Comment by stgeorges** [comment]
I made a [new pull request in here](https://github.com/mostaphaRoudsari/ladybug/pull/279), with all versions changed to 0.0.63.
Changes were made to Photovoltaics and Solar water heating components, as well as Terrain shading mask and Horizon angles (sent to ```7 | WIP```). I added the .py and .ghuser files for the Pedestrian wind comfort component too.

**Comment by chriswmackey** [comment]
Merged! I like the final arrangement of the components and I think it helps to put the PV Surface and Hot Water surface in the first category.

Now we can close this one out!

**Comment by stgeorges** [comment]
Thank you.
